L2/gamma_testnet_fdg_upgrade.md

```diff
@@ -0,0 +1,90 @@
+0. Prepare a pre-funded `$prefunded_pk` to be used for deploying contracts below.
+1. Upgrade ASR：
+    ```bash
+    pushd optimism/packages/contracts-bedrock
+    # deploy UpgradeAnchorStateRegistry contract
+    forge create scripts/deploy/UpgradeAnchorStateRegistry.s.sol:UpgradeAnchorStateRegistry \
+            --broadcast \
+            --private-key $prefunded_pk \
+            --rpc-url $L1_RPC_URL
+    popd
+    cast calldata 'run(address,address,address,address,address,uint32,bytes32,uint256)' \
+        $DISPUTE_GAME_FACTORY_PROXY_ADDRESS $OP_PROXY_ADMIN_ADDRESS \
+        $ANCHOR_STATE_REGISTRY_PROXY_ADDRESS $SUPERCHAIN_CONFIG_PROXY_ADDRESS \
+        0x0000000000000000000000000000000000000000 \
+        0 0xa892c858b32ddb0d5c7c5a53690a28c3163a4ee21c06f7b6000c3db6a05db108 0
+    Delegate call the UpgradeAnchorStateRegistry contract with above calldata from Safe
```

Comment:
It seems that upgradeAndCall will check if `msg.sender` is admin, but in this case, the `msg.sender` is UpgradeAnchorStateRegistry, so the tx will fail?